modules/core/src/main/java/org/apache/ignite/internal/processors/cache/IgniteCacheOffheapManagerImpl.java

```diff
@@ -1546,6 +1498,11 @@ public CacheDataStoreImpl(
             mvccUpdateMarker = new MvccMarkUpdatedHandler(grp);
             mvccUpdateTxStateHint = new MvccUpdateTxStateHintHandler(grp);
             mvccApplyChanges = new MvccApplyChangesHandler(grp);
+
+            if (cleaner == null)
```

Comment:
What is the purpose of change row cache cleaner to a supplier? The code complexity is about the same, the amount of code is about the same, the behavior is about the same.

Reply:
This is done for lazy cache data store initaliztion. Currently, if the page store exists it will trigger the initialization of the cache data store, however, it's not required on partition creation.